File: modules/google-ads-mediation-module/android/src/main/java/expo/modules/googleadsmediationmodule/GoogleAdsMediationModule.kt
package expo.modules.googleadsmediationmodule

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import com.inmobi.sdk.InMobiSdk

class GoogleAdsMediationModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("GoogleAdsMediationModule")

    Function("setInMobiIsAgeRestricted") { hasUserConsent: Boolean ->
      InMobiSdk.setIsAgeRestricted(hasUserConsent)
    }
  }
}

File: modules/google-ads-mediation-module/src/GoogleAdsMediationModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class GoogleAdsMediationModule extends NativeModule<{}> {
  setInMobiIsAgeRestricted(isAgeRestricted: boolean): void;
}

export default requireNativeModule<GoogleAdsMediationModule>('GoogleAdsMediationModule');
